### PR TITLE
Auth extension

### PR DESCRIPTION
**Pull request by sschmittBt**
Added some more options for authentication and login handling

### CODE REVIEW

File: src/main/java/de/braintags/netrelay/controller/authentication/AbstractAuthProviderController.java [hunk 1]
```diff
@@ -84,6 +85,16 @@
   public static final String AUTH_PROVIDER_DATASTORE = "AuthProviderDatastore";
```
Comment (remi128) [inline src/main/java/de/braintags/netrelay/controller/authentication/AbstractAuthProviderController.java:93]: update controller documentation in head ( parameters, usage etc. )

File: src/main/java/de/braintags/netrelay/controller/authentication/AuthenticationController.java [hunk 2]
```diff
@@ -157,12 +160,6 @@
       .getLogger(AuthenticationController.class);
```
Comment (remi128) [inline src/main/java/de/braintags/netrelay/controller/authentication/AuthenticationController.java:164]: dito, add documentation for modified / added parameters into class head

**Review by remi128 — CHANGES_REQUESTED** [review]

**Comment by remi128** [comment]
please add test for JsonForm into TAuthenticationDatastore

File: src/main/java/de/braintags/netrelay/controller/authentication/AbstractAuthProviderController.java (hunk 1 above)
Comment (sschmittBt) [reply src/main/java/de/braintags/netrelay/controller/authentication/AbstractAuthProviderController.java:93]: added documentation

File: src/main/java/de/braintags/netrelay/controller/authentication/AuthenticationController.java (hunk 2 above)
Comment (sschmittBt) [reply src/main/java/de/braintags/netrelay/controller/authentication/AuthenticationController.java:164]: added documentation

**Comment by sschmittBt** [comment]
added unit test

**Pull request merged**